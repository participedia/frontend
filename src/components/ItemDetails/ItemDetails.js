import React from "react";
import auth from "../../utils/AuthService";
import { FormattedMessage } from "react-intl";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { Container, Row, Col } from "reactstrap";
import { FormattedDate } from "react-intl";
import Gallery from "../Gallery";
import BookmarkToggle from "../BookmarkToggle";
import AccordionTab from "../AccordionTab/AccordionTab";
import FloatingActionButton from "material-ui/FloatingActionButton";
import ContentPencil from "material-ui/svg-icons/image/edit";
import caseIconFB from "../../img/pp-case-icon-fb.svg";
import caseIconTW from "../../img/pp-case-icon-tw.svg";
import caseIconLN from "../../img/pp-case-icon-ln.png";
import { ShareButtons } from "react-share";
import htmlToText from "html-to-text";
import "./ItemDetails.css";
import Toggle from "material-ui/Toggle";

function isCurator(profile) {
  if (!profile || !profile.app_metadata || !profile.app_metadata.authorization)
    return false;
  let groups = profile.app_metadata.authorization.groups;
  return groups.indexOf("Curators") !== -1;
}

class Featured extends React.Component {
  static propTypes = {
    toggleFeatured: PropTypes.func.isRequired,
    thing: PropTypes.object.isRequired
  };

  onToggle(object, value) {
    this.props.thing["featured"] = value;
    this.props.toggleFeatured(this.props.thing, value);
  }

  render() {
    const { isAuthenticated } = auth;
    const { profile } = this.props;
    if (!isAuthenticated) return <div />;
    return isCurator(profile) ? (
      <div className="featuretoggle">
        <Toggle
          onToggle={this.onToggle.bind(this)}
          labelPosition="right"
          defaultToggled={this.props.thing.featured}
          label="Featured"
        />
      </div>
    ) : (
      <div />
    );
  }
}

class Hidden extends React.Component {
  static propTypes = {
    toggleHidden: PropTypes.func.isRequired,
    thing: PropTypes.object.isRequired
  };

  onToggle(object, value) {
    this.props.thing["hidden"] = value;
    this.props.toggleHidden(this.props.thing, value);
  }

  render() {
    const { isAuthenticated, profile } = this.props;
    if (!isAuthenticated) return <div />;
    return isCurator(profile) ? (
      <div className="featuretoggle">
        <Toggle
          onToggle={this.onToggle.bind(this)}
          labelPosition="right"
          defaultToggled={this.props.thing.hidden}
          label="Hidden"
        />
      </div>
    ) : (
      <div />
    );
  }
}

const defaultThing = {
  title: "Loading",
  type: "content",
  body: "Description",
  images: [],
  post_date: "2011-11-30T17:20:28",
  updated_date: "2011-11-30T17:20:28",
  location: {
    city: "",
    countrycode: ""
  },
  videos: [],
  creator: { name: "author name" },
  last_updated_by: { name: "author name" }
};

export default class ItemDetails extends React.Component {
  componentWillMount() {
    this.setState({ profile: {} });
    const { isAuthenticated, userProfile, getProfile } = auth;
    if (!userProfile) {
      getProfile((err, profile) => {
        this.setState({ profile, isAuthenticated: isAuthenticated() });
      });
    } else {
      this.setState({
        profile: userProfile,
        isAuthenticated: isAuthenticated()
      });
    }
  }

  render() {
    const {
      FacebookShareButton,
      TwitterShareButton,
      LinkedinShareButton
    } = ShareButtons;
    const intl = this.props.intl;
    const thing = this.props.data || defaultThing;
    let bookmarked = thing.bookmarked;
    let bookmarkIcon = (
      <BookmarkToggle
        thingType={thing.type}
        thingid={thing.id}
        bookmarked={bookmarked}
      />
    );
    let bodyText = htmlToText.fromString(thing.body);
    let currentUrl =
      process.env.REACT_APP_ROOT_URL + this.props.location.pathname;
    let textFacebook = bodyText.substring(0, 240) + "...";
    let creator = thing.creator;
    let creator_url = "/users/" + creator.user_id;
    let creator_name = creator.name;
    let last_updated_by = thing.last_updated_by;
    let last_updated_by_name = last_updated_by.name;
    let last_updated_by_url = "/users/" + last_updated_by.user_id;
    let id = this.props.id;
    let type = thing.type;
    let editLinkUrl = `/${type}/${id}/edit`;
    let detailedBits = React.createElement(this.props.details, {
      case: thing,
      intl
    });

    // XXX L10N -- lots of strings to extract.
    return (
      <div>
        <div className="main-contents">
          <Container className="detailed-case-component" fluid>
            <Row>
              <Col
                md="3"
                className="d-none d-sm-block d-md-block d-lg-block d-xl-block sidepanel"
              >
                <Featured
                  thing={thing}
                  profile={this.state.profile}
                  isAuthenticated={this.state.isAuthenticated}
                  toggleFeatured={this.props.toggleFeatured}
                />
                <Hidden
                  thing={thing}
                  profile={this.state.profile}
                  isAuthenticated={this.state.isAuthenticated}
                  toggleHidden={this.props.toggleHidden}
                />
                {detailedBits}
              </Col>
              <Col md="6" xs="12" className="main-area">
                <div className="case-box">
                  <h3 className="category">
                    <FormattedMessage id={thing.type} />
                  </h3>
                  <h1 className="case-title">{thing.title}</h1>
                  <ul className="icons-mobile clearfix d-md-none d-lg-none d-xl-none">
                    <li>{bookmarkIcon}</li>
                    <li>
                      <FacebookShareButton
                        url={currentUrl}
                        quote={textFacebook}
                      >
                        <img src={caseIconFB} alt="" />
                      </FacebookShareButton>
                    </li>
                    <li>
                      <TwitterShareButton url={currentUrl} title={thing.title}>
                        <img src={caseIconTW} alt="" />
                      </TwitterShareButton>
                    </li>
                    <li>
                      <LinkedinShareButton
                        url={currentUrl}
                        description={textFacebook}
                        title={thing.title}
                      >
                        <img src={caseIconLN} alt="" />
                      </LinkedinShareButton>
                    </li>
                  </ul>
                  <Gallery thing={thing} />
                  <div className="mobile-metadata accordion d-md-none d-lg-none d-xl-none">
                    <AccordionTab titleId={thing.type + "_data"}>
                      <div className="content">{detailedBits}</div>
                    </AccordionTab>
                  </div>
                  <div className="authorship-details">
                    <p className="author-line">
                      First submitted by&nbsp;
                      <Link to={creator_url}>{creator_name}</Link>
                    </p>
                    <p className="date-line">
                      {thing.post_date ? (
                        <FormattedDate
                          value={thing.post_date}
                          year="numeric"
                          month="long"
                          day="2-digit"
                        />
                      ) : (
                        <span />
                      )}
                    </p>
                    <p className="author-line">
                      Most recent changes by&nbsp;
                      <Link to={last_updated_by_url}>
                        {last_updated_by_name}
                      </Link>
                    </p>
                    <p className="date-line">
                      {thing.updated_date ? (
                        <FormattedDate
                          value={thing.updated_date}
                          year="numeric"
                          month="long"
                          day="2-digit"
                        />
                      ) : (
                        <span />
                      )}
                    </p>
                  </div>
                  <div
                    className="case-html"
                    dangerouslySetInnerHTML={{ __html: thing.body }}
                  />
                </div>
              </Col>
              <Col
                md="3"
                className="case-tools d-none d-sm-block d-md-block d-lg-block d-xl-block"
              >
                <div className="top-icons">
                  {bookmarkIcon}
                  <FacebookShareButton url={currentUrl} quote={textFacebook}>
                    <img src={caseIconFB} alt="" />
                  </FacebookShareButton>
                  <TwitterShareButton url={currentUrl} title={thing.title}>
                    <img src={caseIconTW} alt="" />
                  </TwitterShareButton>
                  <LinkedinShareButton
                    url={currentUrl}
                    description={textFacebook}
                    title={thing.title}
                  >
                    <img src={caseIconLN} alt="" />
                  </LinkedinShareButton>
                </div>
              </Col>
            </Row>
          </Container>
          <Link to={editLinkUrl}>
            <FloatingActionButton secondary={true} className="editButton">
              <ContentPencil />
            </FloatingActionButton>
          </Link>
        </div>
      </div>
    );
  }
}
