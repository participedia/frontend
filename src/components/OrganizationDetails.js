import React from "react";
import { FormattedMessage } from "react-intl";

import SearchLink from "../components/SearchLink";
import LinkedPropertyGroupWithHeading from "../components/LinkedPropertyGroupWithHeading";
import ListWithHeading from "../components/ListWithHeading";
import Tags from "./Tags";

export default class OrganizationDetails extends React.Component {
  render() {
    let thing = this.props.case;
    let ed = null;
    if (thing.executive_director) {
      ed = (
        <div>
          <p className="sub-heading">
            <FormattedMessage id="executive_director" />
          </p>
          <div className="sub-sub-heading">{thing.executive_director}</div>
        </div>
      );
    }
    return (
      <div>
        <p className="d-none d-sm-block d-md-block d-lg-block d-xl-block sub-heading">
          <FormattedMessage id="data" />
        </p>
        <Tags thing={thing} />
        <ListWithHeading heading="links" property="links" thing={thing} />
        {ed}

        <LinkedPropertyGroupWithHeading
          heading="sector"
          property="sector"
          thing={thing}
        />
      </div>
    );
  }
}
