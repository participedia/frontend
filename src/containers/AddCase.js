import CaseForm from '../components/CaseForm'
import { reduxForm } from 'redux-form'

const mapStateToProps = (state /* , ownProps */) => {
  return state.form
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    onSubmit: (data) => {
    }
  }
}

export const fields = [ 'firstName', 'lastName', 'age', 'color', 'bio' ]

let AddCaseForm = reduxForm({
  form: 'ncase',
  fields: ['firstName', 'lastName', 'email']
}, mapStateToProps, mapDispatchToProps)(CaseForm)

export default AddCaseForm
