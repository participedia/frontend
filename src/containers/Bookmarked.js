import React from "react";
import api from "../utils/api";
import SearchResults from "./SearchResults";

export default class Bookmarked extends React.Component {
  constructor(props) {
    super(props);
    this.state = { data: null };
  }
  componentDidMount(props) {
    api.getBookmarkedItems().then(data => this.setState({ data }));
  }
  render() {
    if (this.state.data) {
      return <SearchResults data={this.state.data} />;
    } else {
      return <div />;
    }
  }
}
