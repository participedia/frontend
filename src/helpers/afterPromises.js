// shamelessly borrowed from https://github.com/facebook/jest/issues/2136#issuecomment-266817172

export default function afterPromises(done, fn) {
  setTimeout(
    function() {
      try {
        fn();
        done();
      } catch (e) {
        done.fail(e);
      }
    },
    0
  );
}
