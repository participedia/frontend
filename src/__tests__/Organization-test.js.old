import React from "react";
import renderer from "react-test-renderer";

import OrganizationDetails from "../components/OrganizationDetails";
import OrganizationEditor from "../components/OrganizationEditor";
import ItemDetails from "../components/ItemDetails/ItemDetails";
import { IntlProvider } from "react-intl";
import { getBestMatchingMessages } from "../utils/l10n";
let locale = "en-US";
let messages = getBestMatchingMessages(locale);

import { MemoryRouter } from "react-router";
import { Organization } from "../containers/Organization";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import getMuiTheme from "material-ui/styles/getMuiTheme";
const muiTheme = getMuiTheme({});
import { mountWithIntl } from "../helpers/intl-enzyme-test-helper.js";
import data from "./org_data.json";
import intlProps from "../helpers/intl-props-test-helper.js";
import afterPromises from "../helpers/afterPromises";
import injectTapEventPlugin from "react-tap-event-plugin";
injectTapEventPlugin();

let fetchMock = require("fetch-mock");
jest.mock("material-ui/FloatingActionButton");
jest.mock("../components/BookmarkToggle");
jest.mock("react-timeago", () => {
  const React = require("react");
  const TimeAgo = props => React.createElement("TimeAgo");
  return TimeAgo;
}); // so we don't break tests as time goes by
jest.mock("material-ui/TextField", () => "Textfield");
jest.mock("material-ui/RaisedButton");
jest.mock("material-ui/RadioButton");
jest.mock("material-ui/DatePicker");
jest.mock("material-ui/SelectField");
jest.mock("material-ui/SvgIcon");
jest.mock("react-geosuggest");
jest.mock("material-ui-chip-input");
jest.mock("../vendor/react-items-list");
jest.mock("../components/BodyEditor");

fetchMock.get(
  process.env.REACT_APP_API_URL + "/organization/4219",
  JSON.stringify({ data: data })
);
fetchMock.get(
  process.env.REACT_APP_ASSETS_URL + "countries/fullname/Italy.svg",
  JSON.stringify({ data: data })
);

function setup() {
  const props = {
    intl: intlProps,
    auth: { getProfile: cb => cb({}), isAuthenticated: () => false },
    location: { pathname: "/organization/4219" },
    match: { params: { nodeID: 4219 } },
    toggleFeatured: function() {},
    data: data
  };

  const enzymeWrapper = mountWithIntl(
    <MemoryRouter>
      <MuiThemeProvider muiTheme={muiTheme}>
        <Organization {...props} />
      </MuiThemeProvider>
    </MemoryRouter>
  );
  return {
    props,
    enzymeWrapper
  };
}

let props = setup().props;
props["details"] = OrganizationDetails;

describe("containers", () => {
  describe("Organization", () => {
    it("should render proper data for org", done => {
      const { enzymeWrapper } = setup();
      afterPromises(done, () => {
        expect(enzymeWrapper.find("p.author-line").length).toBe(2);
        expect(enzymeWrapper.find("h1.case-title").text()).not.toBe("");
      });
    });
  });
});

test("ItemDetails for Organization renders correctly", () => {
  const tree = renderer
    .create(
      <IntlProvider locale={locale} messages={messages}>
        <MemoryRouter>
          <ItemDetails {...props} toggleHidden={function() {}} />
        </MemoryRouter>
      </IntlProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test("Organization editor renders correctly", () => {
  const tree = renderer
    .create(
      <IntlProvider locale={locale} messages={messages}>
        <MemoryRouter>
          <MuiThemeProvider muiTheme={muiTheme}>
            <OrganizationEditor {...props} type="organization" thing={data} />
          </MuiThemeProvider>
        </MemoryRouter>
      </IntlProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
