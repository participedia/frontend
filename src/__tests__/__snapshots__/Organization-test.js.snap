// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails for Organization renders correctly 1`] = `
<div>
  <div
    className="main-contents"
  >
    <div
      className="detailed-case-component container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="d-none d-sm-block d-md-block d-lg-block d-xl-block sidepanel col col-md-3"
        >
          <div />
          <div />
          <div
            className="d-none d-sm-block d-md-block d-lg-block d-xl-block case-map"
          >
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "",
                }
              }
            />
            <p
              className="case-location"
            >
              Italy
            </p>
          </div>
          <div />
          <div>
            <div
              className="d-none d-sm-block d-md-block d-lg-block d-xl-block case-map"
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "",
                  }
                }
              />
              <p
                className="case-location"
              >
                Italy
              </p>
            </div>
            <p
              className="d-none d-sm-block d-md-block d-lg-block d-xl-block sub-heading"
            >
              <span>
                Data
              </span>
            </p>
            <div>
              <p
                className="sub-sub-heading"
              >
                Tags
              </p>
              <div
                className="tags blond"
              >
                <span>
                  civic engagement
                  ,
                </span>
                <span>
                  digital participation
                  ,
                </span>
                <span>
                  planning
                  ,
                </span>
                <span>
                  public
                </span>
              </div>
            </div>
            <div>
              <p
                className="sub-sub-heading"
              >
                <span>
                  Relevant Links (URLs)
                </span>
              </p>
              <div
                className="links"
              >
                <div>
                  <a
                    href="http://link 1"
                    target="_blank"
                  >
                    link 1
                  </a>
                </div>
                <div>
                  <a
                    href="http://link 2"
                    target="_blank"
                  >
                    link 2
                  </a>
                </div>
                <div>
                  <a
                    href="http://link 3"
                    target="_blank"
                  >
                    link 3
                  </a>
                </div>
              </div>
            </div>
            <div />
          </div>
        </div>
        <div
          className="main-area col-12 col-md-6"
        >
          <div
            className="case-box"
          >
            <h3
              className="category"
            >
              <span>
                Organization
              </span>
            </h3>
            <h1
              className="case-title"
            >
              Urban Interactive Studio
            </h1>
            <ul
              className="icons-mobile clearfix d-md-none d-lg-none d-xl-none"
            >
              <li />
              <li>
                <div
                  className="SocialMediaShareButton SocialMediaShareButton--facebook"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  style={Object {}}
                  tabIndex="0"
                >
                  <img
                    alt=""
                    src="pp-case-icon-fb.svg"
                  />
                </div>
              </li>
              <li>
                <div
                  className="SocialMediaShareButton SocialMediaShareButton--twitter"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  style={Object {}}
                  tabIndex="0"
                >
                  <img
                    alt=""
                    src="pp-case-icon-tw.svg"
                  />
                </div>
              </li>
              <li>
                <div
                  className="SocialMediaShareButton SocialMediaShareButton--linkedin"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  style={Object {}}
                  tabIndex="0"
                >
                  <img
                    alt=""
                    src="pp-case-icon-ln.png"
                  />
                </div>
              </li>
            </ul>
            <div
              className="single-image"
            >
              <img
                alt={undefined}
                className="portrait"
                sizes={undefined}
                src="https://s3.amazonaws.com/uploads.participedia.xyz/uis_favicon_0.png"
                srcSet={undefined}
              />
            </div>
            <div
              className="mobile-metadata accordion d-md-none d-lg-none d-xl-none"
            >
              <div
                className="tab"
                onClick={[Function]}
              >
                <h3
                  className="title"
                >
                  <span>
                    Related Content
                  </span>
                </h3>
                <div
                  className="content"
                >
                  <div />
                </div>
              </div>
              <div
                className="tab"
                onClick={[Function]}
              >
                <h3
                  className="title"
                >
                  <span>
                    Organization Data
                  </span>
                </h3>
                <div
                  className="content"
                >
                  <div>
                    <div
                      className="d-none d-sm-block d-md-block d-lg-block d-xl-block case-map"
                    >
                      <div
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "",
                          }
                        }
                      />
                      <p
                        className="case-location"
                      >
                        Italy
                      </p>
                    </div>
                    <p
                      className="d-none d-sm-block d-md-block d-lg-block d-xl-block sub-heading"
                    >
                      <span>
                        Data
                      </span>
                    </p>
                    <div>
                      <p
                        className="sub-sub-heading"
                      >
                        Tags
                      </p>
                      <div
                        className="tags blond"
                      >
                        <span>
                          civic engagement
                          ,
                        </span>
                        <span>
                          digital participation
                          ,
                        </span>
                        <span>
                          planning
                          ,
                        </span>
                        <span>
                          public
                        </span>
                      </div>
                    </div>
                    <div>
                      <p
                        className="sub-sub-heading"
                      >
                        <span>
                          Relevant Links (URLs)
                        </span>
                      </p>
                      <div
                        className="links"
                      >
                        <div>
                          <a
                            href="http://link 1"
                            target="_blank"
                          >
                            link 1
                          </a>
                        </div>
                        <div>
                          <a
                            href="http://link 2"
                            target="_blank"
                          >
                            link 2
                          </a>
                        </div>
                        <div>
                          <a
                            href="http://link 3"
                            target="_blank"
                          >
                            link 3
                          </a>
                        </div>
                      </div>
                    </div>
                    <div />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="authorship-details"
            >
              <p
                className="author-line"
              >
                First submitted byÂ 
                <a
                  href="/users/undefined"
                  onClick={[Function]}
                >
                  ecrespin
                </a>
              </p>
              <p
                className="date-line"
              >
                <span>
                  May 13, 2015
                </span>
              </p>
              <p
                className="author-line"
              >
                Most recent changes byÂ 
                <a
                  href="/users/undefined"
                  onClick={[Function]}
                >
                  ecrespin
                </a>
              </p>
              <p
                className="date-line"
              >
                <span>
                  May 13, 2015
                </span>
              </p>
            </div>
            <div
              className="case-html"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<h3>Mission and Purpose</h3><p><a href=\\"http://www.urbaninteractivestudio.com\\">Urban Interactive Studio</a> believes that communities should be shaped by people with the assistance of experts, rather than handed down as a finished product. The UIS mission is to give people a voice in shaping the future of their communities.</p><h3>Founder and Philosophy</h3><p>Chris Haller, the CEO and Founder of Urban Interactive Studio,Â is a nationally-recognized User Experience designer and Online Engagement strategist with a broad background in local government, urban and regional planning and communication technologies. In April 2011 he was named one of Planetizen's Top 25 thinkers in Urban Planning Technology, and he brings a sense of creativity and curiosity to all of UIS's clients and projects.</p><p>UIS believes that people and their diverse perspectives are the key to livable communities. But growing use of Web and mobile technology means that traditional methods of public participation like in-person meetings need to be supplemented with digital activities.Â UIS makes these digital activites intuitive so that complex information is easily accessible to citizens and they are in the driverâs seat as they explore and provide feedback on options related the future of their community.</p><p>Urban Interactive Studio also publishes <a href=\\"http://www.engagingcities.com\\">EngagingCities</a>, the prominent online magazine that shares creative strategies and new technologies to foster public engagement for livable communities.</p><h3>Specializations, Tools and Projects</h3><p>Urban Interactive Studio is a cross-disciplinary team of planners, designers and web developers that specializes in providingÂ <strong>intuitive solutions for digital public engagement</strong>. Weâre a small but steadily growing firm at the cutting edge of public process, helping local governments, elected officials, planning and engineering firms, and nonprofits educate, inform and learn from citizens and stakeholders.</p><p>UIS's flagship product,<a href=\\"http://www.urbaninteractivestudio.com/engagingplans\\"> EngagingPlans</a>, is an interactive public engagement platform that enables project teams to facilitate broad-scale public outreach and participation. The EngagingPlansÂ Core includes project updates, event calendars and timelines, a searchable document library, FAQs, a survey builder, polls and open questions.</p><p>The <a href=\\"http://www.urbaninteractivestudio.com/appsuite\\">EngagingPlans App Suite</a> extends Core functionality with a range of apps that offer unique, intuitive opportunities for public involvement and feedback. From community mapping to interactive plan documents, these apps can be tailored to meet the engagement objectives of each project phase.</p><p>UIS has worked with dozens of clients across the country and globe to help connect experts and civic leaders with the local insights necessary to build the best communities possible. Many <a href=\\"http://www.urbaninteractivestudio.com/blog\\">past project stories</a> are chronicled on the blog.</p><h3>External Links</h3><p><a href=\\"http://www.urbaninteractivestudio.com\\">http://www.urbaninteractivestudio.com</a></p><p><a href=\\"http://twitter.com/urbaninteractiv\\">@urbaninteractiv</a></p><p><a href=\\"http://twitter.com/challer\\">@challer</a></p><p><a href=\\"http://www.engagingcities.com\\">http://www.engagingcities.com</a></p><p><a href=\\"http://twitter.com/engagingcities\\">@engagingcities</a></p>",
                }
              }
            />
          </div>
        </div>
        <div
          className="case-tools d-none d-sm-block d-md-block d-lg-block d-xl-block col col-md-3"
        >
          <div
            className="top-icons"
          >
            <div
              className="SocialMediaShareButton SocialMediaShareButton--facebook"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              style={Object {}}
              tabIndex="0"
            >
              <img
                alt=""
                src="pp-case-icon-fb.svg"
              />
            </div>
            <div
              className="SocialMediaShareButton SocialMediaShareButton--twitter"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              style={Object {}}
              tabIndex="0"
            >
              <img
                alt=""
                src="pp-case-icon-tw.svg"
              />
            </div>
            <div
              className="SocialMediaShareButton SocialMediaShareButton--linkedin"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              style={Object {}}
              tabIndex="0"
            >
              <img
                alt=""
                src="pp-case-icon-ln.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <a
      href="/organization/undefined/edit"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Organization editor renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="main-contents"
  >
    <div
      className="detailed-case-component container-fluid"
    >
      <div
        className="d-none d-sm-block d-md-block d-lg-block d-xl-block sidepanel col col-md-3"
      />
      <div
        className="ml-auto mr-auto col col-md-6"
      >
        <div
          className="case-box"
        >
          <div
            className="field-case top"
          >
            <h2
              className="sub-heading"
            >
              <label
                htmlFor="title"
              >
                Title of Organization
              </label>
            </h2>
            <Textfield
              className="custom-field org-title"
              disabled={false}
              errorText={undefined}
              floatingLabelText=""
              fullWidth={true}
              hintText={null}
              name="title"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="organization_title_placeholder"
              type="text"
              value="Urban Interactive Studio"
            />
          </div>
          <div
            className="case-location"
          >
            <div>
              <h3
                className="sub-heading"
              >
                location_placeholder
              </h3>
              <p
                className="explanatory-text"
              >
                Type to search for known location
              </p>
              <div
                className="location"
              />
            </div>
          </div>
          <div
            className="field-case"
          >
            <h2
              className="sub-heading"
            >
              Relevant Links (URLs)
            </h2>
            <div>
              <h3
                className="sub-heading"
              >
                Relevant Links (URLs)
              </h3>
              <p
                className="explanatory-text"
              >
                <span>
                  If there is a main website for this case, enter it here. Add links to additional sources so that readers and editors can find more information about this case online. 
                </span>
              </p>
              <div
                className="list links"
              />
            </div>
          </div>
          <div
            className="field-case media"
          >
            <h2
              className="sub-heading"
            >
              <span>
                Media
              </span>
            </h2>
            <h3
              className="sub-sub-heading"
            >
              <span>
                Photos
              </span>
            </h3>
            <div>
              <div
                className="itemPics row"
              >
                <div
                  className="pl-0 box-container col-12 col-md-6 col-lg-4"
                >
                  <div
                    className="box lead"
                  >
                    <div
                      className="checkbox"
                      onClick={[Function]}
                    />
                    <div
                      className="trash"
                      onClick={[Function]}
                    />
                    <img
                      alt=""
                      className="img-fluid"
                      src="http://uploads.participedia.xyz.s3-website-us-east-1.amazonaws.com/uis_favicon_0.png"
                      style={
                        Object {
                          "height": "100%",
                          "objectFit": "contain",
                          "width": "100%",
                        }
                      }
                    />
                  </div>
                </div>
              </div>
              <p
                className="explanatory-text"
              >
                <span>
                  Select a photo from your device.
                </span>
              </p>
              <div>
                <span>
                  Sorry, upload only works if you're logged in.
                </span>
              </div>
            </div>
            <h3
              className="sub-sub-heading pt-4"
            >
              <span>
                Videos
              </span>
            </h3>
            <div>
              <h3
                className="sub-heading"
              >
                Videos
              </h3>
              <p
                className="explanatory-text"
              >
                <span>
                  Add a video link (Vimeo or YouTube)
                </span>
              </p>
              <div
                className="list videos"
              />
            </div>
          </div>
          <div
            className="field-case"
          >
            <h2
              className="sub-heading"
            >
              Tags
            </h2>
            <div
              className="tags-field"
            >
              <div>
                <h3
                  className="sub-heading"
                >
                  <span>
                    Tags
                  </span>
                </h3>
                <p
                  className="explanatory-text"
                >
                  <span>
                    Select tags that best describe this case.
                  </span>
                </p>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div>
            <div
              className="field-case"
            >
              <h2
                className="sub-heading"
              >
                Executive Director
              </h2>
              <p
                className="explanatory-text"
              >
                executive_director_placeholder
              </p>
              <div
                className="executive_director"
              >
                <Textfield
                  className="custom-field"
                  fullWidth={true}
                  name="executive_director"
                  onChange={[Function]}
                  value=""
                />
              </div>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Sector
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Select Sector",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case select"
            >
              <h3
                className="sub-heading"
              >
                Specific topic
              </h3>
              <p
                className="explanatory-text"
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "specific_topic_instructional",
                    }
                  }
                />
              </p>
            </div>
            <div
              className="field-case"
            >
              <h2
                className="sub-heading"
                htmlFor="body_en"
              >
                Description of the organization
              </h2>
            </div>
            <div
              className="field-case last related"
            >
              <h2
                className="sub-heading"
              >
                Related Content
              </h2>
              <div
                className="related-content"
              >
                <h3
                  className="sub-sub-heading"
                >
                  Related Cases
                </h3>
                <div>
                  <h3
                    className="sub-heading"
                  >
                    <span>
                      Related Cases
                    </span>
                  </h3>
                  <p
                    className="explanatory-text"
                  >
                    <span>
                      Begin typing the title of an existing case
                    </span>
                  </p>
                </div>
                <h3
                  className="sub-sub-heading mt-3"
                >
                  Related Methods
                </h3>
                <div>
                  <h3
                    className="sub-heading"
                  >
                    <span>
                      Related Methods
                    </span>
                  </h3>
                  <p
                    className="explanatory-text"
                  >
                    <span>
                      Begin typing the title of an existing method
                    </span>
                  </p>
                </div>
                <h3
                  className="sub-sub-heading mt-3"
                >
                  Related Organizations
                </h3>
                <div>
                  <h3
                    className="sub-heading"
                  >
                    <span>
                      Related Organizations
                    </span>
                  </h3>
                  <p
                    className="explanatory-text"
                  >
                    <span>
                      Begin typing the title of an existing organization
                    </span>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
